### DISCUSSION TITLE
why can't use 16 channel binary sensors?

### DISCUSSION THREAD

**Discussion by hzkincony** [jpcornil-git/HA-mcp23017#19] [General]
i can use 16 chanenl of OUTPUT, but why can't add 16 channel INPUT?

**Comment by hzkincony** [comment]
it showed error:  Invalid config for 'binary_sensor' from integration 'mcp23017' at configuration.yaml, line 17: 'hw_sync' is an invalid option for 'mcp23017.binary_sensor', check: hw_sync, please check the docs at https://github.com/jpcornil-git/HA-mcp23017

**Reply by jpcornil-git** [reply]
Can you add thé corresponding section of your configuration.yaml file ?

**Reply by jpcornil-git** [reply]
I guess that you used "hw_sync" within your "binary_sensor" definition but this only a valid option for "switch" -> you should remove hw_sync option from your "binary_sensor" definition.

Difficult to say more without more information ...

**Comment by jpcornil-git** [comment]
No further feedbacks/reactions (issue solved i guess) => closing the discussion

**Comment by hzkincony** [comment]
Hello!

yes, it solved, thank you very much.

**Comment by jpcornil-git** [comment]
Thanks for the feedback/confirmation :-) !
